patina_dxe_core/src/image.rs

```diff
@@ -978,6 +992,18 @@ pub fn core_load_image(
     // authenticate the image
     let security_status = authenticate_image(file_path, &image_to_load, boot_policy, from_fv, authentication_status);
 
+    // If a security violation occurs, we still load the image, but will ultimately return a ImageStatus::SecurityViolation
+    if let Err(err) = security_status
+        && err != EfiError::SecurityViolation
+    {
+        // If the error is AccessDenied, we abort loading completely, as platform policy prohibits the image from being loaded
+        if err == EfiError::AccessDenied {
+            return Ok(ImageStatus::AccessDenied);
```

Comment:
aesthetic nit: it's a bit weird to have `Ok(failure)` status. Consider `Result<efi::Handle, ImageStatus>`? where the `Ok` state corresponds is the current `Loaded`?

Ultimately, it isn't a functional issue, so okay if you feel like it is better this way and want to keep it.

Reply:
My argument to stick with this implementation is that the error return case indicates an actual unexpected runtime error where as the OK case is the current status of the image, ejther loaded, loaded with a security violation, or completely blocked by platform policy. While EDKII treats them as EFI_ERRORs, I see them as valid, expected return cases.